src/main/java/org/kiwiproject/registry/eureka/client/EurekaParser.java

```diff
@@ -0,0 +1,84 @@
+package org.kiwiproject.registry.eureka.client;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.Objects.nonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.collections4.MapUtils.getString;
+
+import lombok.experimental.UtilityClass;
+import org.kiwiproject.registry.eureka.common.EurekaInstance;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@UtilityClass
+public class EurekaParser {
+
+    @SuppressWarnings("unchecked")
+    public static List<EurekaInstance> parseEurekaResponse(Map<String, Object> response) {
+        var applications = ((Map<String, Object>) response.get("applications")).get("application");
+
+        var eurekaInstances = new ArrayList<EurekaInstance>();
+
+        if (nonNull(applications)) {
+            if (applications instanceof List<?>) {
+                for (Map<String, Object> application : (List<Map<String, Object>>) applications) {
+                    var instances = application.get("instance");
+                    eurekaInstances.addAll(parseInstances(instances));
+                }
+            } else {
+                eurekaInstances.addAll(parseInstances(((Map<String, Object>) applications).get("instance")));
+            }
+        }
+
+        return eurekaInstances;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static List<EurekaInstance> parseInstances(Object instanceOrInstanceList) {
+        if (instanceOrInstanceList instanceof List<?>) {
+            var instanceList = (List<Map<String, Object>>) instanceOrInstanceList;
+            return instanceList.stream()
+                    .map(EurekaParser::buildInstance)
+                    .collect(toList());
+        }
+
+        var instance = (Map<String, Object>) instanceOrInstanceList;
+        return newArrayList(buildInstance(instance));
+    }
+
+    @SuppressWarnings("unchecked")
+    public static EurekaInstance buildInstance(Map<String, Object> instanceData) {
+
```

Comment:
Here and several other methods...should we do a bit more argument checking? For example this method and the two methods above this one? Here we could just do a `KiwiPreconditions.checkArgumentNotNull(instanceData)` whereas above in `parseInstances` we might want to check-before-cast on line 47.

I suggest creating a separate issue to see where we might benefit from adding some arg checks.